pkg/util/serviceaccount.go

```diff
@@ -83,7 +83,25 @@ func WaitForServiceAccountSecretCreation(client kubeclient.Interface, asObj *cor
 			}
 			return false, fmt.Errorf("failed to retrieve service account(%s/%s) from cluster, err: %v", asObj.Namespace, asObj.Name, err)
 		}
-		clusterSecret, err = GetTargetSecret(client, serviceAccount.Secrets, corev1.SecretTypeServiceAccountToken, asObj.Namespace)
+		if len(serviceAccount.Secrets) == 0 {
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: serviceAccount.Namespace,
+					Name:      serviceAccount.Name,
+					Annotations: map[string]string{
+						"kubernetes.io/service-account.name": serviceAccount.Name,
+					},
+				},
+				Type: corev1.SecretTypeServiceAccountToken,
+			}
+			clusterSecret, err = CreateSecret(client, secret)
```

Comment:
Is it possible that the token information is not generated for the newly created secret?